.gitlab-ci.yml

```diff
@@ -45,22 +46,27 @@ default:
 
 .build:
   stage: build
-  except: [ tags ]
+  rules:
+    - if: '$CI_COMMIT_TAG == null'
+    # If the commit was made on main then use the classic ECR
+    - if: '$CI_COMMIT_BRANCH == "main"'
+      variables:
+        ECR_TEST_ONLY: ""
```

Comment:
Rules are [evaluated in order](https://docs.gitlab.com/ee/ci/yaml/#rules), and only the first matching rule is used. Since `$CI_COMMIT_BRANCH == "main"` => `$CI_COMMIT_TAG == null`, the second case will never get matched and `ECR_TEST_ONLY` will always be `_test_only`.